site-src/geps/gep-1016.md

```diff
@@ -0,0 +1,813 @@
+	//   one hostname that matches the Listener hostname. For example,
+	//   `test.example.com` and `*.example.com` would both match. On the other
+	//   hand, `example.com` and `test.example.net` would not match.
+	//
+	// If both the Listener and GRPCRoute have specified hostnames, any
+	// GRPCRoute hostnames that do not match the Listener hostname MUST be
+	// ignored. For example, if a Listener specified `*.example.com`, and the
+	// GRPCRoute specified `test.example.com` and `test.example.net`,
+	// `test.example.net` must not be considered for a match.
+	//
+	// If both the Listener and GRPCRoute have specified hostnames, and none
+	// match with the criteria above, then the GRPCRoute is not accepted. The
+	// implementation must raise an 'Accepted' Condition with a status of
+	// `False` in the corresponding RouteParentStatus.
+	//
+	// Support: Core
+	//
+	// +optional
+	// +kubebuilder:validation:MaxItems=16
+	Hostnames []Hostname `json:"hostnames,omitempty"`
+
+	// Rules are a list of GRPC matchers, filters and actions.
+        // 
+	// +optional
+	// +kubebuilder:validation:MaxItems=16
+	// +kubebuilder:default={{matches: {{method: {type: "Exact"}}}}}
+	Rules []GRPCRouteRule `json:"rules,omitempty"`
+}
+
+// GRPCRouteRule defines semantics for matching an gRPC request based on
+// conditions (matches), processing it (filters), and forwarding the request to
+// an API object (backendRefs).
+type GRPCRouteRule struct {
+	// Matches define conditions used for matching the rule against incoming
+	// gRPC requests. Each match is independent, i.e. this rule will be matched
+	// if **any** one of the matches is satisfied.
+	//
+	// For example, take the following matches configuration:
+	//
+	// ```
+	// matches:
+	// - method:
+	//     method: foo.bar
+	//   headers:
+	//     values:
+	//       version: 2
+	// - method:
+	//     service: foo.bar.v2
+	// ```
+	//
+	// For a request to match against this rule, a request should satisfy
+	// EITHER of the two conditions:
+	//
+	// - service of foo.bar AND contains the header `version: 2`
+	// - service of foo.bar.v2
+	//
+	// See the documentation for GRPCRouteMatch on how to specify multiple
+	// match conditions that should be ANDed together.
+	//
+	// If no matches are specified, the matches every gRPC request.
+	//
+	// Proxy or Load Balancer routing configuration generated from GRPCRoutes or HTTPRoutes
+	// MUST prioritize rules based on the following criteria, continuing on
+	// ties. When comparing GRPCRoutes and HTTPRoutes, GRPCRoute matches are first converted
+	// to HTTPRoute matches according to the following scheme:
+	//
+	// * HeaderMatches are translated exactly, since the field are
+	//    one-to-one
+	// * If `MethodMatch.type` is `Exact` and neither the
+	//    `MethodMatch.service` nor `MethodMatch.method` fields are present,
+	//    translate to a `PathMatch` of `{"type": "PathPrefix", value: "/"}`
```

Comment:
Would exact match on "/" be a better translation here?

Reply:
Removed this section since we're no longer proposing merge between `GRPCRoute` and `HTTPRoute` (but my last `git push` failed -- whoopsie)

Regardless, this semantic still applies. I think an unspecified method and unspecified match should match _all_ gRPC traffic. Do you think there's an issue with this?

An exact match on `/` is not even a valid gRPC+protobuf URI, so I don't think this makes much sense.